### PR TITLE
feat: Support for coverage

### PR DESCRIPTION
**Pull request by zobo**
Experimental, still WIP.  Let me know if you want me to refactor anything.

### CODE REVIEW

**Comment by recca0120** [comment]
I think we should wait until this functionality is completed before considering the refactoring.

**Comment by zobo** [comment]
I'll push some fixes and I'm working on some basic tests.

File: src/extension.ts [hunk 1]
```diff
@@ -94,7 +95,11 @@ export async function activate(context: ExtensionContext) {
     };
     const testRunProfile = ctrl.createRunProfile('Run Tests', TestRunProfileKind.Run, runHandler, true, undefined, true);
     if (extensions.getExtension('xdebug.php-debug') !== undefined) {
-        ctrl.createRunProfile('Debug Tests', TestRunProfileKind.Debug, runHandler, false, undefined, false);
+        ctrl.createRunProfile('Debug Tests', TestRunProfileKind.Debug, runHandler, true, undefined, false);
+    }
+    const coverageProfile = ctrl.createRunProfile('Run with Coverage', TestRunProfileKind.Coverage, runHandler, true, undefined, false); // TODO Continuous
```
Comment (recca0120) [inline src/extension.ts:100]: I think line 100 - 102 should be moved to line 98

File: src/PHPUnit/CommandBuilder/CommandBuilder.ts [hunk 2]
```diff
@@ -30,6 +31,12 @@ export class CommandBuilder {
         return this;
     }
 
+    setExtraArguments(extraArguments: string[]) {
```
Comment (recca0120) [inline src/PHPUnit/CommandBuilder/CommandBuilder.ts:34]: maybe line 22, setArguments change to addArguments, we don't need this method. I'll refactor this later

File: src/extension.ts (hunk 1 above)
Comment (zobo) [reply src/extension.ts:100]: Well no. Coverage depends on Xdebug PHP extension being installed, whereas debugging depends on Xdebug PHP extension _AND_ xdebug.php-debug vscode extension. We could add a test to check if Xdebug is installed in the currently selected PHP.

But I'm currently not doing this even in the debug vscode extension, so...

**Comment by zobo** [comment]
I have found another problem.
Xdebug mode is controlled by `XDEBUG_MODE` environment and `xdebug.mode` ini (either in the file or via `-d`). The env takes precedence so if I have an environment with XDEBUG_MODE set, I cannot override it with this extension.

I need to change the `CommandBuilder` and `TestRunnerProcess` so that `spawn` modifies the `env`. 
Maybe just removing XDEBUG_MODE would be enough... Ideally phpunit should run without xdebug if it is not needed...

**Pull request merged**